packages/dapi/lib/grpcServer/handlers/blockheaders-stream/subscribeToNewBlockHeaders.js

```diff
@@ -1,52 +1,49 @@
-const { BlockHeader, ChainLock } = require('@dashevo/dashcore-lib');
+const { BlockHeader } = require('@dashevo/dashcore-lib');
 const ProcessMediator = require('./ProcessMediator');
 const wait = require('../../../utils/wait');
 const { NEW_BLOCK_HEADERS_PROPAGATE_INTERVAL } = require('./constants');
+const cache = require('./cache');
+const chainlocks = require('./chainlocks');
 
 /**
  * @typedef subscribeToNewBlockHeaders
  * @param {ProcessMediator} mediator
+ * @param {AppMediator} appMediator
  * @param {CoreRpcClient} coreAPI
  * @param {ZmqClient} zmqClient
  */
-function subscribeToNewBlockHeaders(
-  mediator,
+function subscribeToNewBlockHeaders(mediator,
+  appMediator,
   zmqClient,
-  coreAPI,
-) {
-  const cachedHeadersHashes = new Set();
-  let latestChainLock = null;
+  coreAPI) {
+  const pendingHeadersHashes = new Set();
+
+  let lastChainLock;
 
   let isClientConnected = true;
 
   /**
    * @param {Buffer} hash
    */
   const blockHashHandler = (hash) => {
-    cachedHeadersHashes.add(hash.toString('hex'));
+    pendingHeadersHashes.add(hash.toString('hex'));
   };
 
   /**
-   * @param {Buffer} rawChainLock
+   *
+   * @param rawChainLock {Buffer}
    */
   const rawChainLockHandler = (rawChainLock) => {
-    latestChainLock = new ChainLock(rawChainLock);
+    lastChainLock = rawChainLock;
   };
 
-  zmqClient.on(
-    zmqClient.topics.hashblock,
-    blockHashHandler,
-  );
-
-  zmqClient.on(
-    zmqClient.topics.rawchainlock,
-    rawChainLockHandler,
-  );
+  appMediator.on(appMediator.events.hashblock, blockHashHandler);
```

Comment:
Please use uppercase for the EVENTS constants.
We are generally following this pattern.